SortVisualize/SortVisualize/Controllers/StatisticsDataViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  StatisticsDataViewController.swift
+//  SortVisualize
+//
+//  Created by Sergii Lisnyi on 5/10/18.
+//  Copyright © 2018 Sergii Lisnyi. All rights reserved.
+//
+
+import UIKit
+
+class StatisticsDataViewController: UIViewController {
+
+    
+    @IBOutlet weak var changeModelButton: UIButton!
+    @IBOutlet weak var barProgress: UIProgressView!
+    @IBOutlet weak var typeArraysLabel: UILabel!
+    @IBOutlet weak var dataTable: UITableView!
+
+    var modelData: TypeArrayModelProtocol = SimpleArrayModel()
+    var arrayResult: Array<Array<String>> = [[],[]]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        arrayResult = Array(repeating: Array(repeating: "no data", count: modelData.count), count: TypeSortEnum.count)
+   
+        let queue = DispatchQueue.global(qos: .utility)
+        queue.async {
+             self.doSort(forType: [.insert, .selection, .bubble])
+        }
+
+        OperationQueue().addOperation {
+            self.doSort(forType: [.quick, .merge])
+        }
+    }
+    
+    fileprivate func doSort(forType: [TypeSortEnum]) {
+        let stepBarProgress = 1.0 / Float(TypeSortEnum.count)
+        for i in 0..<forType.count {
+            arrayResult[forType[i].rawValue] = StatisticsSortModel.timeSort(sortType: forType[i], dictionary: modelData.dictionary)
+            DispatchQueue.main.sync {
+                self.dataTable.reloadData()
+                barProgress.progress += stepBarProgress
+                barProgress.progress == 1.0 ? (changeModelButton.isEnabled = true) : (changeModelButton.isEnabled = false)
+            }
+        }
+    }
+    
+    fileprivate func changeModel(modelDataType: TypeArrayEnum) -> TypeArrayModelProtocol {
```

Comment:
Maybe in context of your app you are changing smth,  but the method itself doesn't change anything.
`fileprivate func model(of type: TypeArrayEnum) -> TypeArrayModelProtocol` looks better.
You may even make it calculated property of `TypeArrayEnum`.
PS. `TypeArrayEnum` - very bad not descriptive name